connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -260,7 +417,42 @@ public Response putConnectorConfig(final @PathParam("connector") String connecto
 
     @POST
     @Path("/{connector}/restart")
-    @Operation(summary = "Restart the specified connector")
+    @Operation(
+            summary = "Restart the specified connector",
+            parameters = {
+                    @Parameter(in = ParameterIn.QUERY, hidden = true, schema = @Schema(implementation = BooleanSchema.class), name = "forward")
+            }
+    )
+    @ApiResponses(value = {
+            @ApiResponse(
+                    responseCode = "200",
+                    description = "Named connector exists and the server has successfully stopped and begun restarting only the Connector object"
+            ),
+            @ApiResponse(
+                    responseCode = "202",
+                    description = "Named connector exists and the server has successfully and durably recorded the request to stop and " +
+                            "begin restarting at least one failed or running Connector object and Task instances",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ConnectorStateInfo.class))
+            ),
+            @ApiResponse(
+                    responseCode = "204",
+                    description = "Operation succeeded, but there is no content in the response"
+            ),
+            @ApiResponse(
+                    responseCode = "404",
+                    description = "Named connector does not exist"
+            ),
+            @ApiResponse(
+                    responseCode = "409",
+                    description = "A rebalance is needed, forthcoming, or underway while restarting any of the Connector and/or Task objects"
```

Comment:
We should use original descriptions rather than copying from the Confluent docs!